### PR TITLE
Issue8141/support safari extensions

### LINKED ISSUE #8141: Support safari webextensions in runtime-js
# 🙋 feature request

Runtime-js currently has specific support for chrome and firefox extensions. It has logic to generate a url relative to the bundle root.  For Safari this logic is missing, and parcel generates non-functional URLs. (could be considered a bug, depending on how you look at it)

## 🤔 Expected Behavior

When accessing assets in a safari extension, URLs should be formed correctly by getBundleUrl/getBaseUrl/getOrigin.

## 😯 Current Behavior

URLs generated by getBundleUrl/getBaseUrl/getOrigin do not get transformed and are not generally resolvable in a safari extension.

## 💁 Possible Solution

Adapt the regex in bundle-url.js to also support safari extensions.
I wrote a fix here: https://github.com/parcel-bundler/parcel/commit/6b8231771555acc8d12da49d29ed9b97d4f0c165

### PR DESCRIPTION
**Pull request by Kjoep**
# ↪️ Pull Request

Adapt parcel js runtime to also support safari extensions.
Fixes https://github.com/parcel-bundler/parcel/issues/8141

## 💻 Examples

Any webextension build with parcel that uses getBundleUrl/getBaseUrl/getOrigin should now function.

## 🚨 Test instructions

This can only be tested by building a webextension and running it on safari.  Note that to do this, you need to package the webextension as an OSX application using xcode.

In your extension, add an asset (let's say test.png).  Then add code as follows to your content script:

```
    import icon32 from '~/assets/test.png';
    const icon = document.createElement('img');
    icon.src = browser.runtime.getUrl;
    document.querySelector('body').appendChild(icon);
```

this should add the test image to the document body when your extension is applied (i.e. the import resolves to a `safari-web-extension://` url at runtime).

### CODE REVIEW

File: packages/configs/webextension/package.json [hunk 1]
```diff
@@ -23,5 +23,10 @@
   },
   "peerDependencies": {
     "@parcel/core": "^2.5.0"
+  },
+  "@parcel/bundler-default": {
+    "minBundles": 10000000,
+    "minBundleSize": 3000,
+    "maxParallelRequests": 20
```
Comment (101arrowz) [inline packages/configs/webextension/package.json]: We should address this in a separate PR, for now this should probably be removed.

File: packages/runtimes/js/src/helpers/bundle-url.js [hunk 2]
```diff
@@ -14,7 +14,7 @@ function getBundleURL() {
     throw new Error();
   } catch (err) {
     var matches = ('' + err.stack).match(
-      /(https?|file|ftp|(chrome|moz)-extension):\/\/[^)\n]+/g,
+      /(https?|file|ftp|(chrome|moz|safari-web)-extension):\/\/[^)\n]+/g,
```
Comment (101arrowz) [inline packages/runtimes/js/src/helpers/bundle-url.js:17]: To verify, the scheme for Safari extensions is `safari-web-extension`, not `safari-webextension`, correct?

**Review by 101arrowz — COMMENTED** [review]
Looks good for the most part, just a few questions.

File: packages/configs/webextension/package.json (hunk 1 above)
Comment (Kjoep) [reply packages/configs/webextension/package.json]: Indeed.  That's a workaround for https://github.com/parcel-bundler/parcel/issues/8071 that ended up in this one.  Will remove.

File: packages/runtimes/js/src/helpers/bundle-url.js (hunk 2 above)
Comment (Kjoep) [reply packages/runtimes/js/src/helpers/bundle-url.js:17]: Yes.  (we've already done a release based on this patched version, so it's tested in the wild)

**Review by 101arrowz — APPROVED** [review]

**Review by mischnic — APPROVED** [review]

**Pull request merged**